packages/flutter/test/rendering/object_test.dart

```diff
@@ -23,16 +26,33 @@ void main() {
   });
 
   test('detached RenderObject does not do semantics', () {
-    final TestRenderObject renderObject = new TestRenderObject();
+    final TestSemanticRenderObject renderObject = new TestSemanticRenderObject();
     expect(renderObject.attached, isFalse);
     expect(renderObject.describeSemanticsConfigurationCallCount, 0);
 
     renderObject.markNeedsSemanticsUpdate();
     expect(renderObject.describeSemanticsConfigurationCallCount, 0);
   });
+
+  test('RenderObject can capture images of itself', () async {
```

Comment:
Add a test that expects our assertion error if the render object isn't a repaint boundary?

Reply:
we do test asserts sometimes (check for `takeException`), but this point will be moot anyway if you move it to RenderRepaintBoundary.